stackslib/src/chainstate/stacks/boot/pox_4_tests.rs

```diff
@@ -949,6 +955,28 @@ fn pox_3_defunct() {
     info!("Submitting stacking txs with pox3");
     latest_block = peer.tenure_with_txs(&txs, &mut coinbase_nonce);
 
+    info!("Checking that stackers have no STX locked");
+    let balances = balances_from_keys(&mut peer, &latest_block, &keys);
+    assert_eq!(balances[0].amount_locked(), 0);
+    assert_eq!(balances[1].amount_locked(), 0);
+
+    info!("Checking tx receipts, all `pox3` calls should have returned `(err ...)`");
+    let last_observer_block = observer
+        .get_blocks()
+        .last()
+        .unwrap()
+        .clone();
+
+    let receipts = last_observer_block.receipts
+        .iter()
+        .filter(|receipt| match &receipt.result {
+            Value::Response(r) => !r.committed,
+            _ => false,
+        })
+        .collect::<Vec<_>>();
+
+    assert_eq!(receipts.len(), 4);
```

Comment:
Can you make sure that every transaction failed, i.e. `assert_eq!(receipts.len(), txs.len()`?